crates/biome_js_analyze/src/lint/nursery/no_floating_promises.rs

```diff
@@ -432,6 +510,12 @@ fn is_variable_initializer_a_promise(
             func_expr.async_token().is_some()
                 || is_return_type_a_promise(func_expr.return_type_annotation()).unwrap_or(false),
         ),
+        AnyJsExpression::JsNewExpression(js_new_epr) => {
+            let any_js_expr = js_new_epr.callee().ok()?;
+            let js_ident_expr = any_js_expr.as_js_identifier_expression()?;
+            let reference = js_ident_expr.name().ok()?;
+            Some(reference.has_name("Promise"))
```

Comment:
So, there are few things that we need to handle here:
1. The parenthesis. Unfortunately these are valid cases e.g. `return (new Pormise.resolve())`. You can use the method `.omit_parentheses` that can be used from `AnyJsExpression`
2. Global identifiers. A promise can be used from `window.Promise` or `globalThis.Promise`. You can use the utility `global_identifier`
3. Making sure that `Promise` isn't a binding. You need to use the semantic model to check that. Here's ane example
  https://github.com/biomejs/biome/blob/738ad51d654af10d5dd4faf5e8f8bdef5db81f09/crates/biome_js_analyze/src/lint/suspicious/no_prototype_builtins.rs#L209-L214

Reply:
Thank you for the feedback! 
I have updated the code to handle 1 and 2. I will work on 3 tomorrow.

Reply:
> 1. The parenthesis. Unfortunately these are valid cases e.g. return (new Pormise.resolve()). You can use the method .omit_parentheses that can be used from AnyJsExpression
> 2. Global identifiers. A promise can be used from window.Promise or globalThis.Promise. You can use the utility global_identifier
> 3. Making sure that Promise isn't a binding. You need to use the semantic model to check that.

I have updated the code to cover all the cases.  
For 3 (making sure that Promise isn't a binding), creating a function like `is_global_object` for Promise does not work for globally defined variable. Therefore, I decided to use `model.binding` instead.